### PR TITLE
Add delete functionality for nodes

### PR DESCRIPTION
**Pull request by stianjensen**

### CODE REVIEW

File: src/NodeManager.js [hunk 1]
```diff
@@ -43,6 +43,23 @@
       return node;
     }
 
+    deleteNode(id) {
+      var node = this.nodes[id];
+      for (var key in node.outputs) {
```
Comment (sigvef) [inline src/NodeManager.js:51]: Do you also have to loop through and disconnect GraphInputs as well?

File: src/NodeManager.js [hunk 2]
```diff
@@ -43,6 +43,23 @@
...
+        var inputName = key.split('-')[1];
+        this.disconnect(node.outputs[key].id, inputName);
+      }
+      this.editorDom.removeChild(node.domNode);
+      for (var i = 0; i < this.nodeZStack.length; i++) {
+        var domNode = this.nodeZStack[i].domNode;
+        if (domNode == node.domNode) {
```
Comment (sigvef) [inline src/NodeManager.js]: Why compare domNode and not just node?

Comment (stianjensen) [inline src/NodeManager.js]: I think I must have looked at https://github.com/sigvef/texturegen/pull/35/files#diff-562aafa8fb7a1198ca7e8ba1a1123d4dR74 or something. Will fix

**Comment by stianjensen** [comment]
ping

**Comment by sigvef** [comment]
`selectedNode` in NodeManager needs to cleared when a selected node is deleted.

**Comment by stianjensen** [comment]
ping

**Pull request merged**